plugin/store/api.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package store
+
+import (
+	"context"
+
+	"github.com/opencontainers/go-digest"
+	oci "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/ratify-project/ratify-go/internal/common"
+)
+
+// ListReferrersResult represents the result of ListReferrers API
+type ListReferrersResult struct {
+	Referrers []oci.Descriptor
+	NextToken string
+}
+
+// ReferrerStore is an interface that defines methods to query the graph of supply chain content including its related content
+type ReferrerStore interface {
+	// Name is the name of the store
+	Name() string
+
+	// ListReferrers returns the immediate set of supply chain artifacts for the given subject
+	// represented as artifact manifests.
+	ListReferrers(ctx context.Context, subjectReference common.Reference, artifactTypes []string, nextToken string) (ListReferrersResult, error)
```

Comment:
I think `nextToken` is for page query. Do we need comments to mention when should the caller stop querying?